scripts/diagnostics/fold-adjudication.py

```
@@ -0,0 +1,266 @@
+def png_dimensions(path):
+    with path.open("rb") as source:
+        header = source.read(24)
+    if header[:8] != b"\x89PNG\r\n\x1a\n" or header[12:16] != b"IHDR":
+        raise ValueError(f"Not a PNG: {path}")
+    return struct.unpack(">II", header[16:24])
+
+
+def indexed_pages(manifest_path):
+    manifest = load_json(manifest_path)
+    if manifest.get("version") != 3:
+        raise ValueError(f"Expected protocol-v3 manifest: {manifest_path}")
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)